src/content_sao.h

```diff
@@ -106,31 +106,38 @@ class LuaEntitySAO : public UnitSAO
 {
 public:
 	LuaEntitySAO(ServerEnvironment *env, v3f pos,
-	             const std::string &name, const std::string &state);
+		const std::string &name, const std::string &state);
 	~LuaEntitySAO();
 	ActiveObjectType getType() const
 	{ return ACTIVEOBJECT_TYPE_LUAENTITY; }
 	ActiveObjectType getSendType() const
 	{ return ACTIVEOBJECT_TYPE_GENERIC; }
 	virtual void addedToEnvironment(u32 dtime_s);
 	static ServerActiveObject* create(ServerEnvironment *env, v3f pos,
-			const std::string &data);
+		const std::string &data);
 	void step(float dtime, bool send_recommended);
 	std::string getClientInitializationData(u16 protocol_version);
 	bool isStaticAllowed() const
 	{ return m_prop.static_save; }
 	void getStaticData(std::string *result) const;
 	int punch(v3f dir,
-			const ToolCapabilities *toolcap=NULL,
-			ServerActiveObject *puncher=NULL,
-			float time_from_last_punch=1000000);
+		const ToolCapabilities *toolcap=NULL,
+		ServerActiveObject *puncher=NULL,
+		float time_from_last_punch=1000000);
 	void rightClick(ServerActiveObject *clicker);
 	void setPos(const v3f &pos);
 	void moveTo(v3f pos, bool continuous);
 	float getMinimumSavedMovement();
 	std::string getDescription();
 	void setHP(s32 hp, const PlayerHPChangeReason &reason);
 	u16 getHP() const;
+
+	bool isImmortal()
+	{
+		return itemgroup_get(getArmorGroups(), "immortal") ||
+				!g_settings->getBool("enable_damage");
```

Comment:
I don't know if this works well either.
There may be value in allowing entities to be damaged even when players can't be.

Reply:
Agreed.